### PR TITLE
TE 288 do a cleanup on the svn if the local repository is locked by a…

### PR DESCRIPTION
**Pull request by georgportwich**
… broken command

### CODE REVIEW

File: core/org.testeditor.core/src/main/java/org/testeditor/core/services/impl/TestProjectServiceImpl.java [hunk 1]
```diff
@@ -367,8 +398,8 @@ private File getDirectoryOfDemoProjects(String dir) throws IOException {
 						TestProjectConfig config = getProjectConfigFor(demoProjectName);
 						internalStoreProjectConfig(demoProjectName, config, true);
 					}
-					TestProject project = createProjectFrom(
-							new File(wsDir.getAbsolutePath() + File.separator + demoProjectDir.getName()));
+					TestProject project = createProjectFrom(new File(wsDir.getAbsolutePath() + File.separator
```
Comment (franzbecker) [inline core/org.testeditor.core/src/main/java/org/testeditor/core/services/impl/TestProjectServiceImpl.java:401]: I'm not a big fan of constructing instances of `File` like this. I know it's used all over the place, but why don't we use `new File(wsDir, demoProjectDir.getName())` here instead?

**Comment by franzbecker** [comment]
It seems to me that you're having different formatter settings. The default is 120 characters per line. Could you please reformat the classes?

**Comment by kpanier** [comment]
The core integration test fails.

File: core/org.testeditor.core/src/main/java/org/testeditor/core/services/impl/TestProjectServiceImpl.java [hunk 2]
```diff
@@ -199,10 +222,19 @@ public void unBind(TestServerService testServerService) {
 	private void loadProjectListFromFileSystem() throws SystemException {
 		testProjects = new ArrayList<TestProject>();
 		File[] files = getWorkspaceDirectories();
-		LOGGER.info("Scanning for Projects in: " + Platform.getLocation().toOSString());
+		logger.info("Scanning for Projects in: " + Platform.getLocation().toOSString());
 		for (File file : files) {
 			if (isTestProjectDirectory(file)) {
 				TestProject testProject = createProjectFrom(file);
+				try {
+					if (teamShareService.hasGlobalLock(testProject)) {
```
Comment (kpanier) [inline core/org.testeditor.core/src/main/java/org/testeditor/core/services/impl/TestProjectServiceImpl.java:230]: Other services like the TestStructureService checks that an object is team shared before it calls the team service. this is based on the older implementation without the dispatcher. In this implementation the dispatcher is responsible to handle the no service is defined state. I think we should make this more clear if we decide this responsibility to the dispatcher. Maybe in a new user story.
What do you think?

File: core/org.testeditor.core/src/main/java/org/testeditor/core/services/interfaces/TeamShareService.java [hunk 3]
```diff
@@ -239,4 +239,27 @@ void rename(TestStructure testStructure, String newName, TranslationService tran
...
+
+	/**
+	 * Executes a cleanup for the local repository containing the project.
+	 * 
+	 * @param testProject
+	 *            - a testProject
+	 * @throws SystemException
+	 *             - if there is a problem during cleanup.
+	 */
+	void cleanUp(TestProject testProject) throws SystemException;
```
Comment (franzbecker) [inline core/org.testeditor.core/src/main/java/org/testeditor/core/services/interfaces/TeamShareService.java:263]: cleanup is a single word and should, in my opinion, not use camel case

File: core/org.testeditor.core/src/main/java/org/testeditor/core/services/impl/TestProjectServiceImpl.java (hunk 2 above)
Comment (orhan-polat) [inline core/org.testeditor.core/src/main/java/org/testeditor/core/services/impl/TestProjectServiceImpl.java:230]: To check if a project is  teamshared is a good pattern. New story for this issue is a good idea.

**Pull request merged**